go-controller/pkg/ovn/pods.go

```diff
@@ -109,43 +108,23 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 		klog.Errorf(err.Error())
 	}
 
-	for _, podIPNet := range portInfo.ips {
-		// delete src-ip cached route to GR
-		nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
-		if err != nil {
-			klog.Errorf("Unable to check port: %s, ip: %s for external gw route deletion: %v", portInfo.name,
-				podIPNet.IP, err)
-			continue
-		}
-		podIP := podIPNet.IP.String()
-		if gwToGr, ok := nsInfo.podExternalRoutes[podIP]; ok {
-			if len(gwToGr) == 0 {
-				delete(nsInfo.podExternalRoutes, pod.Status.PodIP)
-				nsInfo.Unlock()
-				continue
-			}
-			mask := GetIPFullMask(podIP)
-			for gw, gr := range gwToGr {
-				_, stderr, err = util.RunOVNNbctl("--", "--if-exists", "--policy=src-ip",
-					"lr-route-del", gr, podIP+mask, gw)
-				if err != nil {
-					klog.Errorf("Unable to delete external gw ecmp route to GR router, stderr:%q, err:%v", stderr, err)
-				} else {
-					delete(nsInfo.podExternalRoutes, pod.Status.PodIP)
-				}
-			}
-		}
-		if config.Gateway.DisableSNATMultipleGWs && nsInfo.routingExternalGWs == nil {
-			nodeName := pod.Spec.NodeName
-			gr := "GR_" + nodeName
-			stdout, stderr, err := util.RunOVNNbctl("--", "--if-exists", "lr-nat-del",
-				gr, "snat", podIP)
-			if err != nil {
-				klog.Errorf("Failed to delete SNAT rule for pod on gateway router %s, "+
-					"stdout: %q, stderr: %q, error: %v", gr, stdout, stderr, err)
-			}
+	podHasMultipleGWs := false
+	routingPods := oc.getRoutingPodGWs(pod.Namespace)
+	if len(routingPods) > 0 {
+		// assume we need to clean up if we cannot find the info
+		podHasMultipleGWs = true
```

Comment:
I don't understand the comment... if `len(routingPods) > 0` then you _could_ find the info. (or is the `>` supposed to be `==`?) (likewise below)

Reply:
yeah this comment and code doesn't make much sense. I think I need to keep the error return on getRoutingPodGWs and getRoutingExternalGWs and check if we couldn't get namespace vs we had got a return value with no pods/gws. I'll fix it.

Reply:
Actually thinking about this some more, there's no reason to even check here. We cant guarantee the namespace still exists when we delete, and we need to clean up anyway ( per the comment). So we just always need to clean up.